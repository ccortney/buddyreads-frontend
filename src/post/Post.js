import { Grid, Button, Typography, Box } from "@mui/material";
import { useState, useContext, useEffect } from "react";
import BuddyReadApi from "../api/api";
import Like from "./Like";
import UserContext from "../auth/UserContext";

const Post = ({post, progress}) => {
    const {currentUser} = useContext(UserContext);
    const [liked, setLiked] = useState(post.liked);
    const [viewed, setViewed] = useState(post.viewed);
    // const [progress, setProgress] = useState();

    // useEffect(() => {
    //     async function getProgress() {
    //         let statRes = await BuddyReadApi.getBuddyReadStat(post.buddyreadId, currentUser.id);
    //         setProgress(statRes.progress)
    //     }
    //     getProgress()
    // }, [])


    const changeLike = async () => {
        await BuddyReadApi.updatePost(post.id, {liked: !liked})
        setLiked(!liked)
    }
    const changeViewed = async () => {
        await BuddyReadApi.updatePost(post.id, {viewed: true})
        setViewed(true)
    }

    return (
        <Box>
            {currentUser.id === post.userId || viewed ? 
                <Typography>
                    {post.user.firstName}'s Post for Page {post.page}: {post.message}
                    <Like changeLike={changeLike} liked={liked} post={post}/> 
                </Typography>
                :
                <Typography>
                    {post.user.firstName}'s Post for Page {post.page}:
                    {post.page <= progress.currentUserProgress ? 
                        <Button onClick={changeViewed}>Reveal Post</Button>
                    :
                        <Button disabled>Reveal Post</Button>
                    } 
                </Typography>
            }

            
        </Box>
    )
}

export default Post;