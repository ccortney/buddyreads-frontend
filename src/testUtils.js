import UserContext from "./auth/UserContext";

const demoUser = {
  id: 0,
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
};

const UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
