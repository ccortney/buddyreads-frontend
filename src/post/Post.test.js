import { render } from "@testing-library/react";
import { UserProvider } from "../testUtils";
import { MemoryRouter } from "react-router";
import Post from "./Post";

const post = {
    id: 1,
    buddyreadId: 1, 
    userId: 1, 
    user: {
        id: 1, 
        firstName: 'First', 
        lastName: 'Last'
    }, 
    page: 1, 
    message: 'message1'
}

const progress = {
    currentUserProgress: 5, 
    buddyProgress: 10
}

it("renders without crashing", function() {
    render(
        <MemoryRouter>
            <UserProvider>
                <Post post={post} progress={progress}/>
            </UserProvider>
        </MemoryRouter>
    );
  });

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
        <UserProvider>
            <Post post={post} progress={progress}/>
        </UserProvider>
    </MemoryRouter>
);;
  expect(asFragment()).toMatchSnapshot();
});