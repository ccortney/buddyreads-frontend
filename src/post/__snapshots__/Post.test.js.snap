// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-187ww8x-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardHeader-root css-1kml5hp-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
      >
        <div
          aria-label="post 1"
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1ow0mo5-MuiAvatar-root"
        >
          F
        </div>
      </div>
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
        >
          First Last
        </span>
        <span
          class="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-83ijpv-MuiTypography-root"
        >
          Page 1
        </span>
      </div>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-10r5to9-MuiCardActions-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1qstl0x-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Reveal Post
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
